src/app/components/goal-dashboard/goal-dashboard.component.ts

```diff
@@ -1559,17 +1518,17 @@ export class GoalDashboardComponent implements OnInit {
      this.dayCommit = this.program_ex_map_name3
      this.dayCommitInitial = new Map(this.program_ex_map_name_in3)
 
-     this.dayOne = this.mthreeDate.getFullYear()+"-"+this.mthreeDate.getMonth()+"-"+this.mthreeDate.getDate()
-     this.dayTwo = this.mtwoDate.getFullYear()+"-"+this.mtwoDate.getMonth()+"-"+this.mtwoDate.getDate()
-     this.dayThree = this.oneDate.getFullYear()+"-"+this.oneDate.getMonth()+"-"+this.oneDate.getDate()
-     this.dayFour = this.twoDate.getFullYear()+"-"+this.twoDate.getMonth()+"-"+this.twoDate.getDate()
-     this.dayFive = this.threeDate.getFullYear()+"-"+this.threeDate.getMonth()+"-"+this.threeDate.getDate()
-     this.daySix = this.fourDate.getFullYear()+"-"+this.fourDate.getMonth()+"-"+this.fourDate.getDate()
-     this.daySeven = this.fiveDate.getFullYear()+"-"+this.fiveDate.getMonth()+"-"+this.fiveDate.getDate()
+     this.dayOne = this.mthreeDate.getFullYear()+"-"+(this.mthreeDate.getMonth()+1)+"-"+this.mthreeDate.getDate()
+     this.dayTwo = this.mtwoDate.getFullYear()+"-"+(this.mtwoDate.getMonth()+1)+"-"+this.mtwoDate.getDate()
+     this.dayThree = this.oneDate.getFullYear()+"-"+(this.oneDate.getMonth()+1)+"-"+this.oneDate.getDate()
+     this.dayFour = this.twoDate.getFullYear()+"-"+(this.twoDate.getMonth()+1)+"-"+this.twoDate.getDate()
+     this.dayFive = this.threeDate.getFullYear()+"-"+(this.threeDate.getMonth()+1)+"-"+this.threeDate.getDate()
+     this.daySix = this.fourDate.getFullYear()+"-"+(this.fourDate.getMonth()+1)+"-"+this.fourDate.getDate()
+     this.daySeven = this.fiveDate.getFullYear()+"-"+(this.fiveDate.getMonth()+1)+"-"+this.fiveDate.getDate()
     }
 
    //---for day 4
-   else if (this.days===3 && this.hours===23 && this.minutes===59 && this.seconds>0){
+   else if (this.days===3 && this.hours>0 && this.minutes>0 && this.seconds>0){
 
      if(this.days===3 && this.hours===23 && this.minutes===59 && this.seconds===59){
        localStorage.removeItem("daily_finish")
```

Comment:
Og det er derfor jeg ikke er fan af hardcodet løsning og gentagen kode. Der er stor fare for at vi overser noget her.